.github/workflows/actions.yml

```
@@ -120,8 +120,15 @@ jobs:
         uses: actions/download-artifact@v3
         with:
           name: poetry-build-artifacts
-      - name: Install package
-        run: ${{ matrix.installer }} install '${{ matrix.installee }}'
+      - name: Pip install package
+        if: matrix.installer == 'pip'
+        run: |
+          ${{ matrix.installer }} install --extra-index-url https://pypi.fury.io/arrow-nightlies/ --prefer-binary --pre pyarrow
+          ${{ matrix.installer }} install '${{ matrix.installee }}'
+      - name: Pipx install package
+        if: matrix.installer == 'pipx'
+        run: |
+          ${{ matrix.installer }} install --pip-args "--extra-index-url https://pypi.fury.io/arrow-nightlies/ --prefer-binary --pre" '${{ matrix.installee }}'
```

```suggestion
          ${{ matrix.installer }} install --pip-args "--extra-index-url https://pypi.fury.io/arrow-nightlies/ --prefer-binary --pre=pyarrow" '${{ matrix.installee }}'
```

Does this work to restrict pyarrow to be the only prerelease allowed?

---

I tried a different approach; instead of the general `--pre` (removed), I added a dev version as minimum (`'pyarrow>=10.0.0dev497'`).

---

After a few tries, it finally works, by just adding the nightly repo to the install commands and then the new minimum version sorts it out automatically.